pallets/elections/src/lib.rs

```diff
@@ -130,44 +160,60 @@ pub mod pallet {
     }
 
     /// Desired size of a committee in effect from a new era.
-    ///
-    /// can be changed via `change_validators` call that requires sudo.
     #[pallet::storage]
     pub type NextEraCommitteeSize<T> =
         StorageValue<_, CommitteeSeats, ValueQuery, DefaultNextEraCommitteeSize<T>>;
 
-    /// List of reserved validators in force from a new era.
-    ///
-    /// Can be changed via `change_validators` call that requires sudo.
+    /// Next era's list of reserved validators.
     #[pallet::storage]
     pub type NextEraReservedValidators<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
 
     /// Current era's list of reserved validators.
-    ///
-    /// This is populated from `NextEraReservedValidators`
-    /// at the time of planning the first session of the era.
-    /// This is a tuple of vectors representing `(reserved, non_reserved)` validators.
     #[pallet::storage]
     pub type CurrentEraValidators<T: Config> =
         StorageValue<_, EraValidators<T::AccountId>, ValueQuery>;
 
-    /// List of possible validators that are not reserved.
+    /// Next era's list of non reserved validators.
     #[pallet::storage]
     pub type NextEraNonReservedValidators<T: Config> =
         StorageValue<_, Vec<T::AccountId>, ValueQuery>;
 
-    /// Count per validator, how many blocks did the validator produced.
+    /// A lookup how many blocks a validator produced.
     #[pallet::storage]
     pub type SessionValidatorBlockCount<T: Config> =
         StorageMap<_, Twox64Concat, T::AccountId, BlockCount, ValueQuery>;
 
     /// Total possible reward per validator for the current era.
-    ///
-    /// Scaled to fit in the u32.
     #[pallet::storage]
     pub type ValidatorEraTotalReward<T: Config> =
         StorageValue<_, ValidatorTotalRewards<T::AccountId>, OptionQuery>;
 
+    /// Default value for kick out config, see [`CurrentEraCommitteeKickOutConfig`]
+    #[pallet::type_value]
+    pub fn DefaultCommitteeKickOutConfig<T: Config>() -> CommitteeKickOutConfig {
+        CommitteeKickOutConfig::default()
+    }
+
+    /// Current era config for kick-out functionality, see [`CommitteeKickOutConfig`]
+    #[pallet::storage]
+    pub type CurrentEraCommitteeKickOutConfig<T> =
```

Comment:
This is because we follow a convention in pallet elections so and also in pallet staking so that each user action performed in the current era (such as an intent to become a validator or adding a new node to non-reserved for example) actually applies in the next era. We could have made a single `CommitteeKickOutConfig` but then we'll lose that functionality.

Reply:
This "convention" makes sense only for very specific cases. We must use it for consensus-critical data that we query in the host one session ahead, such as session authorities or emergency finalizer. Otherwise it is not necessary. Here it's just a config that is not queried anywhere in consensus code.

Reply:
Discussed offline, I'll rework to have only one instance